### PR TITLE
Added a documentation for the Cythonize function and its arguments.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
There might be other possible arguments. I didn't look much. I'm going to do another  PR later with the arguments covered in ``**options``.

### CODE REVIEW

File: docs/src/reference/compilation.rst [hunk 1]
```diff
@@ -237,6 +237,80 @@ Just as an example, this adds ``mylib`` as library to every extension::
...
+                           exclude_failures=False, \
+                           **options)
+
+    Compile a set of source modules into C/C++ files and return a list of distutils
+    Extension objects for them.
+
+    :param module_list: As module list, pass either a glob pattern, a list of glob
+                        patterns or a list of Extension objects.  The latter
+                        allows you to configure the extensions separately
+                        through the normal distutils options.
```
Comment (scoder) [inline docs/src/reference/compilation.rst]: It's even cooler than that! :)
You can also pass `Extension` objects that have `glob` patterns as their `sources`. Then, `cythonize` will resolve the pattern and create a copy of the  `Extension` for every matching file.

File: docs/src/reference/compilation.rst [hunk 2]
```diff
@@ -237,6 +237,80 @@ Just as an example, this adds ``mylib`` as library to every extension::
...
+    :param exclude: When passing glob patterns as ``module_list``, you can exclude certain
+                    module names explicitly by passing them into the ``exclude`` option.
+
+    :param nthreads: The number of concurrent builds for parallel compilation
+                     (requires the Python module multiprocessing).
+
+    :param aliases: If you want to use compiler directives like ``# distutils: ...`` but
+                    can only know at compile time (when running the ``setup.py``) which values
+                    to use, you can use aliases and pass a dictionary mapping those aliases
+                    to Python strings when calling :func:`cythonize`. As an example, sat you
```
Comment (scoder) [inline docs/src/reference/compilation.rst]: sat -> say

File: docs/src/reference/compilation.rst [hunk 3]
```diff
@@ -237,6 +237,80 @@ Just as an example, this adds ``mylib`` as library to every extension::
...
+    :param module_list: As module list, pass either a glob pattern, a list of glob
+                        patterns or a list of Extension objects.  The latter
+                        allows you to configure the extensions separately
+                        through the normal distutils options.
+
+    :param exclude: When passing glob patterns as ``module_list``, you can exclude certain
+                    module names explicitly by passing them into the ``exclude`` option.
+
+    :param nthreads: The number of concurrent builds for parallel compilation
+                     (requires the Python module multiprocessing).
```
Comment (scoder) [inline docs/src/reference/compilation.rst]: requires the \`\`multiprocessing\`\` module.

File: docs/src/reference/compilation.rst [hunk 4]
```diff
@@ -237,6 +237,80 @@ Just as an example, this adds ``mylib`` as library to every extension::
...
+    :param force: Forces the recompilation of the Cython modules, even if the timestamps
+                  don't indicate that a recompilation is necessary.
+
+    :param language: To globally enable C++ mode, you can pass ``language='c++'``. Otherwise, this
+                     will be determined at a per-file level based on compiler directives.  This
+                     affects only modules found based on file names.  Extension instances passed
+                     into :func:`cythonize` will not be changed. It is recommended to rather
+                     use the compiler directive ``# distutils: language = c++`` than this option.
+                     If you don't, Cython will print a message telling you to use the
+                     compiler directives.
```
Comment (scoder) [inline docs/src/reference/compilation.rst]: The last sentence seems redundant. They'll see the message when they disobey the recommendation.

File: docs/src/reference/compilation.rst [hunk 5]
```diff
@@ -237,6 +237,80 @@ Just as an example, this adds ``mylib`` as library to every extension::
...
+
+    :param exclude_failures: For a broad 'try to compile' mode that ignores compilation
+                             failures and simply excludes the failed extensions,
+                             pass ``exclude_failures=True``. Note that this only
+                             really makes sense for compiling ``.py`` files which can also
+                             be used without compilation.
+
+    :param annotate: If ``True``, will produce a HTML file for each of the ``.pxd`` files compiled.
```
Comment (scoder) [inline docs/src/reference/compilation.rst]: ".pxd files" -> ".pyx or .py files"

File: docs/src/reference/compilation.rst [hunk 6]
```diff
@@ -237,6 +237,80 @@ Just as an example, this adds ``mylib`` as library to every extension::
...
+                     This HTML file will show each line of Cython code with how much
+                     Python interaction there is. It also allows you to see the C/C++ code
+                     generated for each line of Cython code. This report is invaluable when
+                     optimizing a function for speed,
+                     and for determining when to :ref:`release the GIL <nogil>`:
+                     in general, a ``nogil`` block may contain only "white" code.
+                     See examples in :ref:`determining_where_to_add_types` or
+                     :ref:`primes`.
```
Comment (scoder) [inline docs/src/reference/compilation.rst]: Nice!

File: docs/src/reference/compilation.rst [hunk 7]
```diff
@@ -237,6 +237,80 @@ Just as an example, this adds ``mylib`` as library to every extension::
...
+
+    :param exclude_failures: For a broad 'try to compile' mode that ignores compilation
+                             failures and simply excludes the failed extensions,
+                             pass ``exclude_failures=True``. Note that this only
+                             really makes sense for compiling ``.py`` files which can also
+                             be used without compilation.
+
+    :param annotate: If ``True``, will produce a HTML file for each of the ``.pxd`` files compiled.
+                     This HTML file will show each line of Cython code with how much
+                     Python interaction there is. It also allows you to see the C/C++ code
```
Comment (scoder) [inline docs/src/reference/compilation.rst]: "The HTML file gives an indication of how much Python interaction there is in each of the source code lines, compared to plain C code."

File: docs/src/reference/compilation.rst (hunk 1 above)
Comment (gabrieldemarmiesse) [reply docs/src/reference/compilation.rst]: Fixed next line.

File: docs/src/reference/compilation.rst (hunk 6 above)
Comment (gabrieldemarmiesse) [reply docs/src/reference/compilation.rst]: I can't fix that.

**Pull request merged**